src/main/java/uk/gov/pay/products/resources/PaymentResource.java

```diff
@@ -63,7 +63,7 @@ public PaymentResource(PaymentsFactory paymentsFactory, ProductsFactory products
     @PermitAll
     public Response findPaymentsByProductExternalId(@PathParam("productId") String productExternalId){
         logger.info("Find a list of payments with product id - [ {} ]", productExternalId);
-        Optional<Integer> productId = productsFactory.productsFinder().findProductIdByExternalId(productExternalId);
+        Optional<Integer> productId = productFactory.productsFinder().findProductIdByExternalId(productExternalId);
         if(!productId.isPresent()){
```

Comment:
This line and the following could be replaced with 
```
return productId.map(integer -> {
            List<Payment> payments = paymentsFactory.paymentsFinder().findByProductId(productId.get());
            return payments.size() > 0
                    ? Response.status(OK).entity(payments).build()
                    : Response.status(NOT_FOUND).build();

        })
                .orElseGet(() -> Response.status(NOT_FOUND).build());
```
To keep in line with the pattern from the previous methods.
`map()` will apply the check first as `if (!isPresent())`

Reply:
I have changed although we have to be careful with functional idiom and try not abuse of them at the expense of readability.

Reply:
Agreed